client/http/http.go

```diff
@@ -119,10 +119,10 @@ func ForURLs(urls []string, chainHash []byte) []client.Client {
 	return clients
 }
 
-func Ping(ctx context.Context, root string) error {
+func Ping(root string) error {
 	url := fmt.Sprintf("%s/health", root)
 
-	ctx, cancel := context.WithCancel(ctx)
+	ctx, cancel := context.WithTimeout(context.Background(), maxTimeoutHTTPRequest)
```

Comment:
is there a reason the caller wouldn't want to pass in a context here? it would be nice if we don't need to change the exposed interface here, and it seems like we could wrap the passed in context with `WithTimeout`

Reply:
Good point, let's not change the API.